tests/Splices.hs

```diff
@@ -10,15 +10,15 @@ $(either error return $ parseDecs
 
 -- Testing that the [] constructor works in types, 
 $(either error return $ parseDecs 
-      "nilp :: [a] -> ([] a)\nnilp [] = []")
+      "nilp :: forall a. [a] -> ([] a)\nnilp [] = []")
```

Comment:
I'm not sure if this is the best way to fix the error:

```
tests/Splices.hs:12:3: Not in scope: type variable ‘a’
```
I'm not sure what's going on here. `-ddump-splices` output:

```
tests/Splices.hs:(12,3)-(13,43): Splicing declarations
    either error return
    $ parseDecs
        "nilp :: [a] -> ([] a)\n\
        \nilp [] = []"
  ======>
    nilp :: [a] -> ghc-prim-0.4.0.0:GHC.Types.[] a
    nilp [] = []
```

which is the same for GHC >= 8 as well (except GHC 8.6 [Trac #13776](https://ghc.haskell.org/trac/ghc/ticket/13776)).

Reply:
I don't think changing the test suite is the way to go here (clearly the program is type correct and should be translated to type correct code). The type error did not appear on GHC 8.2 when I tested it.

Reply:
Likely related to b3305d289589e7ba9531f2aa9d977142c89778b5